### ISSUE TITLE
✨ Allow layer list view to be more flexible

### ISSUE THREAD

**Issue by liaprins-czi** [napari/napari#3291]
### Overview of design need
- **Is there an existing GitHub issue this design work pertains to?** 
[More than two layers in the layerlist should be visible without scrolling on small screens #1710](https://github.com/napari/napari/issues/1710)
[Heuristic: Layer panels layout #18](https://github.com/napari/product-heuristics-2020/issues/18)
[Heuristic: Layer controls relationships #19](https://github.com/napari/product-heuristics-2020/issues/19)

- In a sentence or two, describe the problem to be solved for users
There is often extra, unused space below the layer controls, and meanwhile the layers list is getting cut off, forcing user to scroll to see more than two layers in the list. Ideally the heights of the layer controls and layer list would adjust or would be adjustable, or else might be set a little higher so more layers can be shown and less blank layer control area (but need to verify the varying heights of layer controls, per layer type).

### What level of design is needed? (Choose all that apply)
- [ ] **User experience research:**  high-level recommendation/exploration of user needs, design heuristics, and / or best practices to inform a design experience  
(Use this option when you feel there’s a challenge to be solved, but you’re curious about what the experience should be — may involve research studies to understand challenges/opportunities for design)
- [ ] **information flow / conceptual:** organizing and structuring of information flow and content, including layout on screen or across multiple steps
- [X] **Visual:** creating mockups, icons, etc
(If choosing this level alone, it means that the content to be mocked up and its organization is already known and specified)

### Is design a blocker?
- [ ] **Yes:** engineering cannot proceed without a design first
- [X] **No:** engineering can create a first version, and design can come in later to iterate and refine

If selecting **Yes**, how much design input is needed to unblock engineering? For example, is a full, final visual design needed, or just a recommendation of which conceptual direction to go?

### T-shirt size
- [X] ●○○○○ XS: able to be done in a typical workday or two
- [ ] ●●○○○ S: able to do within one sprint, if also working on other things in the sprint
- [ ] ●●●○○ M: able to do within one sprint, if it is the sole focus
- [ ] ●●●●○ L: will likely take more than one sprint to complete, but less than two
- [ ] ●●●●● XL: a very large experience to be determined or overhauled, will need to be broken down further

**Comment by tlambert03** [comment]
> Ideally the heights of the layer controls and layer list would adjust or would be adjustable,

I think one thing we definitely want to avoid here is "reflow" every time someone selects a different layer.  If the layer controls start having variable height based on the number of widgets in them, then it the layers themselves would "jump around" as you selected different ones.  If the layer controls were below the layers, that might work, but I'm not sure that feels as natural...  thoughts?

**Comment by liaprins-czi** [comment]
Yeah jumping of top of layer list top would not be ideal. I'm not sure about having the layer controls below the list or not. It's hard to tell if I'm just used to it being how it is now. But one argument for teh controls being below the layers list is that they are dependent upon the layer selected. So it could actually be more intuitive if they controls aer lower, since they are a dependency on what is selected (could be above them). But not sure. Ideally the user would ahve the control to change the height of each section relative to each other... @tlambert03 or @goanpeca (or others), do you know if it's possible to make the height of layer controls relative to height of layers list adjustable in UI by user, while they are in panel form?

**Comment by tlambert03** [comment]
> @tlambert03 or @goanpeca (or others), do you know if it's possible to make the height of layer controls relative to height of layers list adjustable in UI by user, while they are in panel form?

pretty much everything is possible! :)  the ratio of dock widget heights can be adjusted with [`QtMainWindow.resizeDocks`](https://doc.qt.io/qt-5/qmainwindow.html#resizeDocks)
